utils.go

```diff
@@ -66,3 +81,55 @@ func IsBinary(data []byte) bool {
 
 	return true
 }
+
+func loadGitattributes() (map[string]string, error) {
+	gitAttributes := map[string]string{}
+	data, err := ioutil.ReadFile(".gitattributes")
+	if err != nil {
+		if err != os.ErrNotExist {
+			log.Println(".gitattributes: " + err.Error())
+		}
+		return nil, err
+	}
+
+	if data != nil {
+		tokens := strings.Fields(string(data))
+		for i := 0; i < len(tokens); i = i + 2 {
```

Comment:
Maybe you should use a scanner or split before by lines, it's not so clear what are you doing here at a glance